libs/src/sdk/services/Storage/Storage.ts

```diff
@@ -19,15 +19,18 @@ import {
 } from './constants'
 import type { CrossPlatformFile as File } from '../../types/File'
 import { isNodeFile } from '../../utils/file'
+import { StorageNodeSelector } from './StorageNodeSelector'
 
 export class Storage implements StorageService {
   /**
    * Configuration passed in by consumer (with defaults)
    */
   private readonly config: StorageServiceConfig
+  private readonly storageNodeSelector: StorageNodeSelector
 
-  constructor(config?: StorageServiceConfig) {
+  constructor(config: StorageServiceConfig) {
     this.config = mergeConfigWithDefaults(config, defaultStorageServiceConfig)
+    this.storageNodeSelector = new StorageNodeSelector(this.config)
```

Comment:
It feels like this should take an injected `StorageNodeSelector` as a constructor arg, pref some interface that can be mocked as opposed to the impl as the type

Reply:
that's fine with me too, just thinking config wise it might be easier for folks to not have to DI everything since i dont see this used anywhere else